src/build/__init__.py

```diff
@@ -142,6 +142,7 @@ def check_dependencies(self, distribution):  # type: (str) -> Set[str]
         :param distribution: Distribution to build (sdist or wheel)
         '''
         dependencies = self.get_dependencies(distribution)
+        dependencies.update(self.build_dependencies)
```

Comment:
Do we still need this considering @pfmoore issues was a typo?

Reply:
It is still needed. Without it, there's no warning/error in `--no-isolation` mode if a build requirement is missing (the build will probably fail, of course, but the point of my report was that I think an up-front check is better).